x-pack/plugin/migrate/src/internalClusterTest/java/org/elasticsearch/xpack/migrate/action/ReindexDatastreamIndexIT.java

```diff
@@ -0,0 +1,408 @@
+                }
+            }
+        }
+        """;
+
+    public void testTsdbStartEndSet() throws Exception {
+        assumeTrue("requires the migration reindex feature flag", REINDEX_DATA_STREAM_FEATURE_FLAG.isEnabled());
+
+        var templateSettings = Settings.builder().put("index.mode", "time_series");
+        if (randomBoolean()) {
+            templateSettings.put("index.routing_path", "metricset");
+        }
+        var mapping = new CompressedXContent(TSDB_MAPPING);
+
+        // create template
+        var request = new TransportPutComposableIndexTemplateAction.Request("id");
+        request.indexTemplate(
+            ComposableIndexTemplate.builder()
+                .indexPatterns(List.of("k8s*"))
+                .template(new Template(templateSettings.build(), mapping, null))
+                .dataStreamTemplate(new ComposableIndexTemplate.DataStreamTemplate(false, false))
+                .build()
+        );
+        client().execute(TransportPutComposableIndexTemplateAction.TYPE, request).actionGet();
+
+        // index doc
+        Instant time = Instant.now();
+        String backingIndexName;
+        {
+            var indexRequest = new IndexRequest("k8s").opType(DocWriteRequest.OpType.CREATE);
+            indexRequest.source(TSDB_DOC.replace("$time", formatInstant(time)), XContentType.JSON);
+            var indexResponse = client().index(indexRequest).actionGet();
+            backingIndexName = indexResponse.getIndex();
+        }
+
+        var sourceSettings = indicesAdmin().getIndex(new GetIndexRequest().indices(backingIndexName))
+            .actionGet()
+            .getSettings()
+            .get(backingIndexName);
+        Instant startTime = IndexSettings.TIME_SERIES_START_TIME.get(sourceSettings);
+        Instant endTime = IndexSettings.TIME_SERIES_END_TIME.get(sourceSettings);
+
+        // sanity check start/end time on source
+        assertNotNull(startTime);
+        assertNotNull(endTime);
+        assertTrue(endTime.isAfter(startTime));
+
+        // force a rollover so can call reindex and delete
+        var rolloverRequest = new RolloverRequest("k8s", null);
+        var rolloverResponse = indicesAdmin().rolloverIndex(rolloverRequest).actionGet();
+        rolloverResponse.getNewIndex();
+
+        // call reindex on the original backing index
+        var destIndex = client().execute(ReindexDataStreamIndexAction.INSTANCE, new ReindexDataStreamIndexAction.Request(backingIndexName))
+            .actionGet()
+            .getDestIndex();
+
+        var destSettings = indicesAdmin().getIndex(new GetIndexRequest().indices(destIndex)).actionGet().getSettings().get(destIndex);
+        var destStart = IndexSettings.TIME_SERIES_START_TIME.get(destSettings);
+        var destEnd = IndexSettings.TIME_SERIES_END_TIME.get(destSettings);
+
+        assertEquals(startTime, destStart);
+        assertEquals(endTime, destEnd);
+    }
+
+    // TODO more logsdb/tsdb specific tests
```

Comment:
There was no logsdb/tsdb in 7.x was there? So nothing to upgrade?

Reply:
Oh sweet. Well, then I can remove some tests